src/components/app/App.jsx

```diff
@@ -1,49 +1,38 @@
-import React, { useState } from 'react';
-
-const useRecord = (init) => {
-  const [before, setBefore] = useState([]);
-  const [current, setCurrent] = useState(init);
-  const [after, setAfter] = useState([]);
-
-  const undo = () => {
-    setAfter((after) => [current, ...after]);
-    setCurrent(before[before.length - 1]);
-    setBefore((before) => before.slice(0, -1));
-  };
-
-  const redo = () => {
-    setBefore((before) => [...before, current]);
-    setCurrent(after[0]);
-    setAfter((after) => after.slice(1));
-  };
-
-  const record = (val) => {
-    setBefore((before) => [...before, current]);
-    setCurrent(val);
-  };
-
-  return {
-    undo,
-    record,
-    redo,
-    current,
-  };
-};
+/* eslint-disable max-len */
+import React from 'react';
+import { useDispatch, useSelector } from '../../state/RecordProvider';
+import { undoAction, redoAction, recordAction } from '../../state/actions';
+
 
 function App() {
-  const { current, undo, redo, record } = useRecord('#FF0000');
+  const dispatch = useDispatch();
+  const selector = useSelector();
+
+  const undo = () => (
+    dispatch(undoAction())
+  );
+
+  const redo = () => (
+    dispatch(redoAction())
+  );
+
+  const record = value => (
+    dispatch(recordAction(value))
+  );
 
   return (
     <>
       <button onClick={undo}>undo</button>
       <button onClick={redo}>redo</button>
       <input
         type="color"
-        value={current}
+        value={selector.current}
```

Comment:
Close!
You'll want to create a file called selector.js or something like that 
```
export const selectCurrent = state => state.current;

export const selectBefore = state => state.before;

export const selectAfter = state => state.after;

```

then you'll bring in those into App.js and use them like 

```
  const current = useSelector(selectCurrent);
  const before = useSelector(selectBefore);
  const after = useSelector(selectAfter);
```